routers/api/v1/user/app.go

```diff
@@ -128,15 +131,41 @@ func DeleteAccessToken(ctx *context.APIContext) {
 	//   required: true
 	// - name: token
 	//   in: path
-	//   description: token to be deleted
-	//   type: integer
-	//   format: int64
+	//   description: token to be deleted, identify by ID or name
+	//   type: string
 	//   required: true
 	// responses:
 	//   "204":
 	//     "$ref": "#/responses/empty"
+	//   "422":
+	//     "$ref": "#/responses/error"
+
+	token := ctx.Params(":id")
+	var tokenID int64
+
+	if match, err := regexp.MatchString("^\\d+$", token); err != nil || !match {
+		tokens, err := models.ListAccessTokens(models.ListAccessTokensOptions{
+			Name:   token,
+			UserID: ctx.User.ID,
+		})
+		if err != nil {
+			ctx.Error(http.StatusInternalServerError, "ListAccessTokens", err)
+		}
+
+		switch len(tokens) {
+		case 0:
+			ctx.NotFound()
+			return
+		case 1:
+			tokenID = tokens[0].ID
+		default:
+			ctx.Error(http.StatusUnprocessableEntity, "DeleteAccessTokenByID", fmt.Errorf("multible matches for token name '%s'", token))
+			return
+		}
+	} else if tokenID, err = strconv.ParseInt(token, 0, 64); err != nil {
+		ctx.Error(http.StatusInternalServerError, "ParseInt", err)
+	}
```

Comment:
OK two options,  this is where token name overrides token ID:

```suggestion
	tokenID, _ := strconv.ParseInt(token, 0, 64)
	
	tokens, err := models.ListAccessTokens(models.ListAccessTokensOptions{
		Name: token,
		UserID: ctx.User.ID,
	})
	if err != nil {
		ctx.Error(http.StatusInternalServerError, "ListAccessTokens", err)
		return
	}

	switch len(tokens) {
	case 0:
		// fallback to the strconv parsed
	case 1:
		tokenID = tokens[0].ID
	default:
		ctx.Error(http.StatusUnprocessableEntity, "DeleteAccessTokenByID", fmt.Errorf("multible matches for token name '%s'", token))
		return
	}
	
	if tokenID == 0 {
		ctx.Error(http.StatusInternalServerError, fmt.Sprintf("Unable to parse %s as integer", token), nil)
		return
	}
```

This is where Token ID overrides Token name:

```suggestion
	tokenID, _ := strconv.ParseInt(token, 0, 64)

	if tokenID == 0 {		
		tokens, err := models.ListAccessTokens(models.ListAccessTokensOptions{
			Name: token,
			UserID: ctx.User.ID,
		})
		if err != nil {
			ctx.Error(http.StatusInternalServerError, "ListAccessTokens", err)
			return
		}

		switch len(tokens) {
		case 0:
			ctx.NotFound()
			return
		case 1:
			tokenID = tokens[0].ID
		default:
			ctx.Error(http.StatusUnprocessableEntity, "DeleteAccessTokenByID", fmt.Errorf("multible matches for token name '%s'", token))
			return
		}
	}
	if tokenID == 0 {
		ctx.Error(http.StatusInternalServerError, "Invalid TokenID", nil)
		return
	}
```